tests/test_validation.py

```diff
@@ -689,17 +689,47 @@ def test_unfrozen_parameters_w_peft_layers_to_transform(self):
         ):
             validate_config(cfg)
 
-    def test_hub_model_id_save_value_warns(self):
-        cfg = DictDefault({"hub_model_id": "test"})
+    def test_hub_model_id_save_value_warns_save_stragey_no(self):
+        cfg = DictDefault({"hub_model_id": "test", "save_strategy": "no"})
+
+        with self._caplog.at_level(logging.WARNING):
+            validate_config(cfg)
+            assert (
+                "set without any models being saved" in self._caplog.records[0].message
+            )
+
+    def test_hub_model_id_save_value_warns_random_value(self):
+        cfg = DictDefault({"hub_model_id": "test", "save_strategy": "test"})
 
         with self._caplog.at_level(logging.WARNING):
             validate_config(cfg)
             assert (
                 "set without any models being saved" in self._caplog.records[0].message
             )
 
-    def test_hub_model_id_save_value(self):
-        cfg = DictDefault({"hub_model_id": "test", "saves_per_epoch": 4})
+    def test_hub_model_id_save_value_steps(self):
+        cfg = DictDefault({"hub_model_id": "test", "save_strategy": "steps"})
+
+        with self._caplog.at_level(logging.WARNING):
+            validate_config(cfg)
+            assert len(self._caplog.records) == 0
+
+    def test_hub_model_id_save_value_epochs(self):
+        cfg = DictDefault({"hub_model_id": "test", "save_strategy": "epoch"})
+
+        with self._caplog.at_level(logging.WARNING):
+            validate_config(cfg)
+            assert len(self._caplog.records) == 0
+
+    def test_hub_model_id_save_value_none(self):
+        cfg = DictDefault({"hub_model_id": "test", "save_strategy": None})
+
+        with self._caplog.at_level(logging.WARNING):
+            validate_config(cfg)
+            assert len(self._caplog.records) == 0
+
+    def test_hub_model_id_save_value_no_set_save_strategy(self):
+        cfg = DictDefault({"hub_model_id": "test"})
```

Comment:
This test is wrong, I believe. It should error as there's no `save_steps` or `saves_per_epoch` or `save_strategy`

Reply:
Not setting save_strategy means it becomes steps as I understand it @NanoCode012 and then the test should pass right?

Reply:
You're right.